src/core/crypto/ecdsa.cpp

```diff
@@ -31,6 +31,16 @@
  *   This file implements ECDSA signing.
  */
 
+/**
+ * Direct access to fields of structures declared in public headers is no longer
+ * supported. In Mbed TLS 3, the layout of structures is not considered part of
+ * the stable API, and minor versions (3.1, 3.2, etc.) may add, remove, rename,
+ * reorder or change the type of structure fields.
+ */
+#if !defined(MBEDTLS_ALLOW_PRIVATE_ACCESS)
+#define MBEDTLS_ALLOW_PRIVATE_ACCESS
+#endif
```

Comment:
Instead of doing that, I recommend using `foo.MBEDTLS_PRIVATE(field)` at the point of access. This way you don't risk accidentally accessing private fields where you didn't intend to.

Reply:
Sure, in fact, I did it first but reverted to the approach used Zephyr's PR. Thanks for the suggestion.